### PR TITLE
task(fxa-settings): add BentoMenu component

### PR DESCRIPTION
**Pull request by meandavejustice**
## This pull request

- Implements the global Bento menu into fxa settings

## Issue that this pull request solves

Closes: #4867 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).

## Screenshots (Optional)

This is the current state of the mobile view:
image

This is the desktop view:
image

This is what the mobile view looks like on https://monitor.mozilla.com (the floating email field is a bug on the monitor site, (https://github.com/mozilla/blurts-server/issues/1951)
image

and monitor desktop:
image

## Other information (Optional)

Here is a checklist of the tasks I need to complete.

- [x] convert the app icons from the current sprite sheet to svgs. I've requested these and am waiting for a link. (this is more of a code quality bug which can be addressed in the next maintenance sprint if we want)
- [x] Make the bento menu fill the full view on mobile, with the close icon. (I wanted verify that we want to follow this pattern in settings (@pwalm), it seems reasonable 2 me. 
- [x] convert the css rules added in `tailwind.scss` to classes, (this is tied to the svg conversion)
- [x] Add a couple more unit tests.
- [x] Finish adding a11y attributes and get a review from someone who is qualified to do such things.
- [x] pop this into the header/nav (note on this below)

I'm opening this draft pr to show the current state of the bento menu, I'll need to switch over to address #4874 ([FXA-1583](https://jira.mozilla.com/browse/FXA-1583))so that it can make it into the current sprint also, and so that we have a good place for the bento menu to fit into the nav.

### CODE REVIEW

**Comment by meandavejustice** [comment]
Found and replaced most of the svgs, will finish those up in the morning

File: packages/fxa-settings/src/components/BentoMenu/index.tsx [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                        <div className="inline-block -mb-1 pr-1">
+                          <LockwiseIcon />
+                        </div>
+                        Firefox Lockwise
+                      </LinkExternal>
+                    </li>
+                  </ul>
+                </div>
+                <LinkExternal
+                  className="link-blue text-xs w-full text-center block mt-2 p-2 hover:bg-grey-100"
```
Comment (LZoog) [inline packages/fxa-settings/src/components/BentoMenu/index.tsx:123]: `m-2` looks a bit closer to Monitor styles than `mt-2`.

File: packages/fxa-settings/src/styles/tailwind.scss [hunk 2]
```diff
@@ -37,3 +38,14 @@ body {
     @apply transition duration-150;
   }
 }
+/* stylelint-disable */
+@responsive {
+  .drop-down-menu {
+    @apply w-64 bg-white absolute shadow-md rounded;
```
Comment (LZoog) [inline packages/fxa-settings/src/styles/tailwind.scss:44]: The Monitor menu is definitely more rounded (maybe `rounded-lg`) but I'd say match whatever Figma is looking like on the avatar menu and we can check with design later.

File: packages/fxa-settings/src/styles/tailwind.scss [hunk 3]
```diff
@@ -5,10 +5,11 @@
 @import './animations.css';
 @import './ctas.scss';
 @import './unit-row.scss';
-@import './drop-down-menu.scss';
+
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+@tailwind screens;
```
Comment (LZoog) [inline packages/fxa-settings/src/styles/tailwind.scss:12]: Is this needed for `@responsive` to work? If so maybe that's why it has to be moved in this file.

File: packages/fxa-settings/src/components/BentoMenu/index.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+          />
+        ) : (
+          <BentoIcon
+            role="img"
+            onClick={toggleRevealed}
+            data-testid="drop-down-bento-menu-toggle"
+            title="Firefox Bento Menu"
+            aria-expanded={isRevealed}
+            aria-controls={dropDownId}
+            className="cursor-pointer p-1 pb-0 focus:outline-none transition-standard w-6 mx-6 desktop:mx-8"
```
Comment (LZoog) [inline packages/fxa-settings/src/components/BentoMenu/index.tsx:43]: This should be inside of a button that gets these `aria` values (see `DropDownAvatarMenu`) which will make that `title` show up (and you can take off `role="img"` if it's in a button with a title). 

It's also missing a hover state.

And, this looks better to me at what would be a new class at `w-7` personally, seems a little small despite matching the help icon? But `w-8` seems too large, maybe double check with design?

File: packages/fxa-settings/src/components/BentoMenu/index.tsx [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+          />
+        )}
+
+        {isRevealed && (
+          <div
+            id={dropDownId}
+            data-testid={dropDownId}
+            className="rounded tablet:drop-down-menu tablet:top-10 tablet:-left-54"
```
Comment (LZoog) [inline packages/fxa-settings/src/components/BentoMenu/index.tsx:51]: The caret looks a little off to me at `-left-54` which is `-13.5rem`:

image


`-12.25rem` seems to center it better:

image

File: packages/fxa-react/configs/storybooks.js [hunk 6]
```diff
@@ -64,7 +64,7 @@ const customizeWebpackConfig = ({ config, mode }) => {
   });
 
   config.module.rules.push({
-    test: /\.(woff|woff2|eot|ttf)$/,
+    test: /\.(woff|woff2|eot|ttf|png|jpe?g|gif)$/,
```
[the remark this answers is not in the document]
Comment (meandavejustice) [reply packages/fxa-react/configs/storybooks.js:67]: This was left over from when I initially built it with the sprite sheet from the monitor server, removing.

File: packages/fxa-settings/src/components/BentoMenu/close.svg [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<svg width="24px" height="24px" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (meandavejustice) [reply packages/fxa-settings/src/components/BentoMenu/close.svg:1]: I thought that is what was in https://mozilla.invisionapp.com/d/main#/console/18744572/397559436/preview, but I'm having trouble logging back into invision. It's not sending me the reset link 🤔

File: packages/fxa-settings/src/components/BentoMenu/index.tsx [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+
+import { ReactComponent as FirefoxIcon } from './logo.svg';
+import { ReactComponent as MonitorIcon } from './monitor.svg';
+import { ReactComponent as PocketIcon } from './pocket.svg';
+import { ReactComponent as DesktopIcon } from './desktop.svg';
+import { ReactComponent as MobileIcon } from './mobile.svg';
+import { ReactComponent as LockwiseIcon } from './lockwise.svg';
+import { ReactComponent as BentoIcon } from './bento.svg';
+import { ReactComponent as CloseIcon } from './close.svg';
+import LinkExternal from 'fxa-react/dist/components/LinkExternal';
```
[the remark this answers is not in the document]
Comment (meandavejustice) [reply packages/fxa-settings/src/components/BentoMenu/index.tsx:17]: 👍 I think that was a "use quick fix" vs code thing, I'll fix it up

File: packages/fxa-settings/src/components/BentoMenu/index.tsx [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+export const BentoMenu = () => {
+  const [isRevealed, setRevealed] = useState(false);
+  const toggleRevealed = () => setRevealed(!isRevealed);
+  const closeFn = () => setRevealed(false);
+  const bentoMenuInsideRef = useClickOutsideEffect<HTMLDivElement>(setRevealed);
+  useEscKeydownEffect(setRevealed);
+  const dropDownId = 'drop-down-bento-menu';
+
+  return (
+    <>
```
[the remark this answers is not in the document]
Comment (meandavejustice) [reply packages/fxa-settings/src/components/BentoMenu/index.tsx:28]: Copy pasta from dropdown component, removing

File: packages/fxa-settings/src/components/BentoMenu/close.svg (hunk 7 above)
[the remark this answers is not in the document]
Comment (meandavejustice) [reply packages/fxa-settings/src/components/BentoMenu/close.svg:1]: I think it looks better with just the bento menu, so I'm removing the close icon

File: packages/fxa-settings/src/styles/tailwind.scss (hunk 2 above)
Comment (meandavejustice) [reply packages/fxa-settings/src/styles/tailwind.scss]: updated to rounded-lg, I think it looks okay on both of them now

File: packages/fxa-settings/src/styles/tailwind.scss (hunk 3 above)
Comment (meandavejustice) [reply packages/fxa-settings/src/styles/tailwind.scss:13]: It's needed both here and the separate file 👍 that's why we were having such weirdness with the responsive targetting 🤦

**Comment by meandavejustice** [comment]
I think I have addressed everything, had some weirdness with the breakpoints which we shouldn't run into in the future since `@tailiwind screens` is now added in the appropriate places.

File: packages/fxa-settings/src/components/BentoMenu/index.tsx (hunk 4 above)
Comment (meandavejustice) [reply packages/fxa-settings/src/components/BentoMenu/index.tsx]: Added button, updated attributes, added hover state, and adjusted width, margin, and padding

File: packages/fxa-settings/src/components/BentoMenu/index.tsx (hunk 5 above)
Comment (meandavejustice) [reply packages/fxa-settings/src/components/BentoMenu/index.tsx]: Cleaned this up and got looked exact

File: packages/fxa-settings/src/components/BentoMenu/close.svg (hunk 7 above)
[the remark this answers is not in the document]
Comment (meandavejustice) [reply packages/fxa-settings/src/components/BentoMenu/close.svg:1]: I've made that change, we are pulling from `fxa-react/images/close.svg` currently on mobile

File: packages/fxa-react/configs/tailwind.js [hunk 10]
```diff
@@ -57,7 +58,11 @@ module.exports = {
         '1/2': '50%',
         '-px': '-1px',
         '-3': '-0.75rem',
+        '-50': '-11.60rem',
```
[the remark this answers is not in the document]
Comment (meandavejustice) [reply packages/fxa-react/configs/tailwind.js:61]: `-11.5rem` isn't actually in the center of the button

**Review by LZoog — APPROVED** [review]
Awesome work here, this looks so good across all breakpoints! 🎉

**Pull request merged**